app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEActivity.java

```diff
@@ -219,102 +199,44 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
             if (event.getAction() == MotionEvent.ACTION_DOWN) {
                 v.requestFocus();
             }
             if (event.getAction() == MotionEvent.ACTION_UP) {
                 hideKeyboard();
                 v.clearFocus();
-                presenter.onFabClick(needsSearch.get());
+                viewModel.onSearchClick();
             }
             return true;
         });
 
-        binding.navigationBar.pageConfiguration(pageConfigurator);
-        binding.navigationBar.setOnNavigationItemSelectedListener(item -> {
-            switch (item.getItemId()) {
-                case R.id.navigation_list_view:
-                    binding.mainLayout.setVisibility(View.VISIBLE);
-                    binding.mainComponent.setVisibility(GONE);
-                    showMap(false);
-                    showSearchAndFilterButtons();
-                    break;
-                case R.id.navigation_map_view:
-                    if (backDropActive) {
-                        closeFilters();
-                    }
-                    binding.mainLayout.setVisibility(View.VISIBLE);
-                    binding.mainComponent.setVisibility(GONE);
-                    showMap(true);
-                    showSearchAndFilterButtons();
-                    break;
-                case R.id.navigation_analytics:
-                    fromAnalytics = true;
-                    if (backDropActive) {
-                        closeFilters();
-                    }
-                    binding.mainComponent.setVisibility(View.VISIBLE);
-                    binding.mainLayout.setVisibility(GONE);
-                    showAnalytics();
-                    hideSearchAndFilterButtons();
-                    break;
-            }
-            return true;
-        });
+        configureBottomNavigation();
+
         try {
             binding.filterRecyclerLayout.setAdapter(filtersAdapter);
         } catch (Exception e) {
             Timber.e(e);
         }
 
-        binding.mapLayerButton.setOnClickListener(view -> {
-            new MapLayerDialog(teiMapManager)
-                    .show(getSupportFragmentManager(), MapLayerDialog.class.getName());
-        });
-
-        binding.mapPositionButton.setOnClickListener(view -> {
-            teiMapManager.centerCameraOnMyPosition((permissionManager) -> {
-                permissionManager.requestLocationPermissions(this);
-                return Unit.INSTANCE;
-            });
-        });
-
         binding.executePendingBindings();
         if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
             setFabVisibility(false, false);
         }
 
         if (savedInstanceState != null) {
-            presenter.restoreQueryData((HashMap<String, String>) savedInstanceState.getSerializable(Constants.QUERY_DATA));
+            viewModel.restoreQueryData((HashMap<String, String>) savedInstanceState.getSerializable(Constants.QUERY_DATA));
```

Comment:
maybe this could be called in the viewModel itself, inside` init {} `method.

Reply:
done!